arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt

```diff
@@ -543,41 +544,61 @@ public sealed class Ior<out A, out B> {
 
   public fun isNotEmpty(): Boolean = !isLeft
 
+  @OptIn(ExperimentalTypeInference::class)
+  @OverloadResolutionByLambdaReturnType
   public inline fun <C> traverse(fa: (B) -> Iterable<C>): List<Ior<A, C>> =
     fold(
       { a -> listOf(Left(a)) },
       { b -> fa(b).map { Right(it) } },
       { a, b -> fa(b).map { Both(a, it) } }
     )
 
+  @Deprecated("traverseEither is being renamed to traverse to simplify the Arrow API", ReplaceWith("traverse(f)"))
   public inline fun <AA, C> traverseEither(fa: (B) -> Either<AA, C>): Either<AA, Ior<A, C>> =
+    traverse(fa)
+
+  @OptIn(ExperimentalTypeInference::class)
+  @OverloadResolutionByLambdaReturnType
+  public inline fun <AA, C> traverse(fa: (B) -> Either<AA, C>): Either<AA, Ior<A, C>> =
     fold(
       { a -> Either.Right(Left(a)) },
       { b -> fa(b).map { Right(it) } },
       { a, b -> fa(b).map { Both(a, it) } }
     )
 
-  public inline fun <C> traverseOption(fa: (B) -> Option<C>): Option<Ior<A, C>> =
+  @OptIn(ExperimentalTypeInference::class)
+  @OverloadResolutionByLambdaReturnType
+  public inline fun <C> traverse(fa: (B) -> Option<C>): Option<Ior<A, C>> =
     fold(
       { a -> Some(Left(a)) },
       { b -> fa(b).map { Right(it) } },
       { a, b -> fa(b).map { Both(a, it) } }
     )
 
+  @Deprecated("traverseOption is being renamed to traverse to simplify the Arrow API", ReplaceWith("traverse(f)"))
```

Comment:
Is it ok that the argument is called `fa` but the replacement just uses `f`?

Reply:
good catch didn't see that, will change it :)